src/Bindings/ManualBindings_World.cpp

```diff
@@ -380,9 +378,9 @@ static int tolua_cWorld_QueueTask(lua_State * tolua_S)
 		return cManualBindings::lua_do_error(tolua_S, "Error in function call '#funcname#': Could not get function reference of parameter #1");
 	}
 
-	auto task = std::make_shared<cLuaWorldTask>(*Plugin, FnRef);
-	Plugin->AddResettable(task);
-	self->QueueTask(task);
+	auto ResettableTask = std::make_shared<cLuaWorldTask>(*Plugin, FnRef);
+	Plugin->AddResettable(ResettableTask);
+	self->QueueTask(std::bind(&cLuaWorldTask::Run, ResettableTask, std::placeholders::_1));
```

Comment:
I agree that this weird hybrid of lambda's and strategy objects is a bad idea. Particularly as your taking a member function pointer and treating it as a function pointer.

What would be a much better way of doing this would be to take advantage of the fact that std::function will take anything that meets the Callable concept. So if you replace the Run method with an override of operator(), you should be able to construct a std::function from it.